src/common/registry/test/validator-registry/registry-update.spec.ts

```diff
@@ -339,3 +324,90 @@ describe('Empty registry', () => {
     await registryService.update(address, blockHash);
   });
 });
+
+describe('Reorg detection', () => {
+  const provider = new JsonRpcBatchProvider(process.env.PROVIDERS_URLS);
+  let registryService: ValidatorRegistryService;
+  let registryStorageService: RegistryStorageService;
+  let moduleRef: TestingModule;
+  const mockCall = jest.spyOn(provider, 'call').mockImplementation(async () => '');
+  const CHAIN_ID = process.env.CHAIN_ID || 1;
+  const address = REGISTRY_CONTRACT_ADDRESSES[CHAIN_ID];
+  let mikroOrm: MikroORM;
+
+  jest.spyOn(provider, 'detectNetwork').mockImplementation(async () => getNetwork('mainnet'));
+
+  beforeEach(async () => {
+    const imports = [
+      MikroOrmModule.forRoot(mikroORMConfig),
+      MockLoggerModule.forRoot({
+        log: jest.fn(),
+        error: jest.fn(),
+        warn: jest.fn(),
+      }),
+      ValidatorRegistryModule.forFeature({ provider }),
+    ];
+    moduleRef = await Test.createTestingModule({
+      imports,
+      providers: [{ provide: LOGGER_PROVIDER, useValue: {} }],
+    }).compile();
+    registryService = moduleRef.get(ValidatorRegistryService);
+    registryStorageService = moduleRef.get(RegistryStorageService);
+    mikroOrm = moduleRef.get(MikroORM);
+    const generator = mikroOrm.getSchemaGenerator();
+    await generator.updateSchema();
+  });
+
+  afterEach(async () => {
+    mockCall.mockReset();
+    await clearDb(mikroOrm);
+    await registryStorageService.onModuleDestroy();
+  });
+
+  test('init on update', async () => {
+    const saveRegistryMock = jest.spyOn(registryService, 'saveOperators');
+    const saveKeyRegistryMock = jest.spyOn(registryService, 'saveKeys');
+    const finalizedUsedSigningKeys = 1;
+
+    const keysWithModuleAddress = keys.map((key) => {
+      return { ...key, moduleAddress: address };
+    });
+
+    const operatorsWithModuleAddress = operators.map((key) => {
+      return { ...key, moduleAddress: address, finalizedUsedSigningKeys };
+    });
+
+    const unrefMock = registryServiceMock(moduleRef, provider, {
```

Comment:
I have a question here, Why we're assigning the result of registryServiceMock to unrefMock. If I'm not mistaken, registryServiceMock is a function and doesn’t return another function, right?  so maybe it is enough call registryServiceMock function

Reply:
and maybe unrefMock()  we dont need below

Reply:
checked again, i see , now it returns reset function